cognitive/src/test/scala/com/microsoft/azure/synapse/ml/services/openai/OpenAIPromptSuite.scala

```diff
@@ -49,10 +49,44 @@ class OpenAIPromptSuite extends TransformerFuzzing[OpenAIPrompt] with OpenAIAPIK
 
   test("Basic Usage JSON") {
     prompt.setPromptTemplate(
-      """Split a word into prefix and postfix a respond in JSON
-        |Cherry: {{"prefix": "Che", "suffix": "rry"}}
-        |{text}:
-        |""".stripMargin)
+        """Split a word into prefix and postfix a respond in JSON
+          |Cherry: {{"prefix": "Che", "suffix": "rry"}}
+          |{text}:
+          |""".stripMargin)
+      .setPostProcessing("json")
+      .setPostProcessingOptions(Map("jsonSchema" -> "prefix STRING, suffix STRING"))
+      .transform(df)
+      .select("outParsed")
+      .where(col("outParsed").isNotNull)
+      .collect()
+      .foreach(r => assert(r.getStruct(0).getString(0).nonEmpty))
+  }
+
+  lazy val promptGpt4: OpenAIPrompt = new OpenAIPrompt()
+    .setSubscriptionKey(openAIAPIKey)
+    .setDeploymentName(deploymentNameGpt4)
+    .setCustomServiceName(openAIServiceName)
+    .setOutputCol("outParsed")
+    .setTemperature(0)
+
+  test("Basic Usage - Gpt 4") {
```

Comment:
nit: it looks like you also add extra functionality to drop or keep the messages col. Perhaps either test this or remove functionality. Also if you keep functionality the user should be able to set what the name of the messages column should be

Reply:
Just added functionality to name the message column and tests testing keeping the message column!